src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFileProvider.cs

```diff
@@ -145,7 +146,7 @@ internal PhysicalFilesWatcher FileWatcher
                     ref _fileWatcher,
                     ref _fileWatcherInitialized,
                     ref _fileWatcherLock,
-                    _fileWatcherFactory);
+                    _fileWatcherFactory) ?? throw new NullReferenceException(nameof(_fileWatcher));
```

Comment:
this should be a separate change - we should avoid any product changes when adding annotations

Reply:
also seems to me this will actually never return null because factory never returns null. I'd just `!` this.

Reply:
This is an edge case. This overload of `EnsureInitialized` will return `null` only if `_fileWatcherInitialized` is `true` and `_fileWatcher` is `null`. This doesn't seem like a valid flow anyway. 
Would you mind removing`_fileWatcherInitialized` altogether?